core/node/base_token_fetcher/src/lib.rs

```diff
@@ -119,11 +128,8 @@ impl BaseTokenFetcher {
 
 #[async_trait]
 impl ConversionRateFetcher for BaseTokenFetcher {
-    fn conversion_rate(&self) -> anyhow::Result<u64> {
-        anyhow::Ok(
-            self.latest_to_eth_conversion_rate
-                .load(std::sync::atomic::Ordering::Relaxed),
-        )
+    fn conversion_rate(&self) -> anyhow::Result<BigDecimal> {
+        Ok(self.latest_to_eth_conversion_rate.lock().unwrap().clone())
```

Comment:
Maybe we could return a reference to avoid cloning.

Reply:
Also, shouldn't `self.latest_to_eth_conversion_rate.lock()` be enough?

Reply:
You have a point here, but the related trait returns a `Result<BigDecimal>`, I think returning a reference here would overly complicate this interface (plus I don't think that's thread-safe), and maybe forcing the implementator to use an Arc<...> or an alternatively thread-safe alternative maybe a bit too much? I think this is ok, as long as it does not hinder performance.

Reply:
Cloning bigdecimal is cheap, so I see no problem here.